Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -202,6 +197,69 @@ private extension UInt8 {
     }
 }
 
+/// Represent the current modification state of the HTTPHeaders.
+fileprivate enum HTTPHeadersModificationState {
+    /// Was modified and the underlying storage is not continous anymore.
+    case modifiedNonContinuous
+    /// Was modified and the underlying storage is continous.
+    case modified
+
+    /// Was not modified and the underlying storage is continous.
+    case notModified
+
+    /// Header was added.
+    mutating func addedHeader() {
+        if case .notModified = self {
+            self = .modified
+        }
+    }
+
+    /// Header was removed.
+    mutating func removedHeader() {
+        guard case .modifiedNonContinuous = self else {
+            self = .modifiedNonContinuous
+            return
+        }
+    }
+
+    /// Underlying storage is continous.
+    var isContinuous: Bool {
+        return self != .modifiedNonContinuous
+    }
+
+    /// HTTPHeaders were modified.
+    var wasModified: Bool {
+        return self != .notModified
+    }
+}
+
+// Keep state of keep alive state.
+internal enum KeepAliveState {
+    // We know keep alive should be used.
+    case keepAlive
+    // We know we should close the connection.
+    case close
+    // We need to scan the headers to find out if keep alive is used or not
+    case needScan(continuous: Bool)
```

Comment:
Why is there a continuous flag here? You have a `HTTPHeadersModificationState` that tracks this.

Reply:
Because I forgot to remove the modification state... it’s all tracked via this one now

Reply:
ok removed

Reply:
I don't like this much. I'd rather this bool was moved out of this enum: it just has nothing to do with what this enum is tracking.

If we do that, we can then name the third case of this enum `unknown`, and then the implementation of `HTTPHeaders.isKeepAlive` gets a lot nicer because it can just return `.unknown` when it doesn't know the answer instead of being passed the HTTP version.